modules/cli/src/main/scala/scala/cli/exportCmd/SbtProjectDescriptor.scala

```diff
@@ -274,10 +274,16 @@ final case class SbtProjectDescriptor(
   private def dependencySettings(options: BuildOptions, scope: Scope): SbtProject = {
 
     val depSettings = {
-      val depStrings = options.classPathOptions
-        .extraDependencies.toSeq.toList
-        .map(_.value)
-        .map { dep =>
+      val allDeps =
+        options.classPathOptions.extraDependencies.toSeq.toList.map(_.value).map((_, false)) ++
+          options.classPathOptions.extraCompileOnlyDependencies
+            .toSeq
+            .toList
+            .map(_.value)
+            .map((_, true))
+
+      val depStrings = allDeps.map {
+        case (dep, isCompileOnly) =>
```

Comment:
MAybe a function instead of  zip?

Something like:
```
def toDepString(isCompileOnly: Boolean)(dep: ...) = 
``
and then just map `extraDependencies` and `extraCompileOnlyDependencies` separately?